nengo_loihi/hardware/snips/nengo_host.cc.template

```diff
@@ -0,0 +1,147 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <iostream>
+#include <string>
+#include <unistd.h>  // usleep
+
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <arpa/inet.h>
+
+#include "nxsdkhost.h"
+
+#define N_OUTPUTS {{ n_outputs }}
+//#define BUFFER_SIZE {{ host_buffer_size }}
+#define BUFFER_SIZE 1024
+
+#define SERVER_PORT htons({{ server_port }})
+
+
+namespace nengo_host {
+
+const char input_channel[] = "{{ input_channel }}";
+const char output_channel[] = "{{ output_channel }}";
+
+class NengoHostProcess : public ConcurrentHostSnip {
+    int32_t buffer[BUFFER_SIZE];
+    int server_socket;
+    int client_socket;
+
+  public:
+    NengoHostProcess() {
+        // --- set up socket to communicate with superhost
+        // This machine will act as the server, and the superhost will
+        // connect in as the client
+        server_socket = socket(AF_INET, SOCK_STREAM, 0);
```

Comment:
For socket programming, I found that it is common to check for errors from functions execution. It might not be needed here probably(not sure). But just in case, the usual checks include if the output of the function is less than 0. e.g.
`if (server_socket < 0)
     std::cout << "Failed to open the socket" << std::endl;
`
The error output is `-1` for `socket`,`bind`,`listen` ,`accept`

Reply:
To add the error statements, what should I be doing after the print statement?
Option 1) `return -1`. Although I haven't been able to figure out who would be calling `NengoHostProcess`  and if returning a particular value would mean anything.
Option 2) have a `while`  loop to ensure the success before moving on, but that would still require a block for `timeout` and a `return` statement

Reply:
Just do what you suggested first: Print an error message to `cout`, and then return (option 1). This is in the constructor, so I'm not sure if you can return `-1`.

Also, I'm worried that even if you return early from the constructor, the host snip `run` function will still get called. `client_socket` gets set right at the end of the constructor, so if we initialize both sockets to `-1` and have an error at the start of `run` if either one is < 0, then we should be good.